### ISSUE TITLE
task: refactor

### ISSUE THREAD

**Issue by mizu-bai** [CQPES/JaxPIP#5]
1. is it necessary to expose Jp(xyz) in descriptor class? also auto-batch by checking input's ndim? additionally, what about the base class for models?
2. if so, during xyz -> p, should we use jax.jacfwd to manually manage the gradients?

```python3
@jax.jit
def descriptor_val_and_grad(xyz):
    p = descriptor(xyz)
    J_p = jax.jacfwd(descriptor)(xyz)

    return p, J_p
```

3. keep the model classes pure, only JIT at the outer evolution loops (DMC & QCT)
4. introduce min-max scalar for PIP-NN
5. `pyproject.toml`
6. cli tools

```bash
# from BAS file generated by MSA-2.0
$ jaxpip bas2json /path/to/bas [/path/to/json] [--gz]
# show number of atoms, flat monomials, and polynomials
$ jaxpip show [jaxpip.json | jaxpip.json.gz]
```

both will show

```
# ------------------------------
# Basis File: water_pip.json.gz
# Number of Atoms:       3
# Number of Monomials:   12
# Number of Polynomials: 48
# ------------------------------
```

**Comment by mizu-bai** [comment]
https://github.com/CQPES/JaxPIP/commit/aa985b69c6b5fed4134c70a38c506be0c32a97c8
